src/crystal/system/file_handle.cr

```diff
@@ -0,0 +1,81 @@
+class Crystal::System::FileHandle
+  include IO
+
+  # Create a new `FileHandle` using the platform-specific representation of this
+  # handle.
+  #
+  # On unix-like platforms this constructor takes an Int32 file descriptor.
+  # def self.new(platform_specific) : FileHandle
+
+  # Returns the platform-specific representation of this handle.
+  #
+  # On unix-like platforms this returns an Int32 file descriptor.
+  # def platform_specific
```

Comment:
`platform_specific` is the worst possible naming. If it's supposed to represent the underlying system data, then maybe `raw` or even `handle` if we go with windows naming (bad idea: Windows is *one* target, POSIX is *all* others) or just a plain `#to_unsafe` so we can pass `@handle` directly to C functions instead of the horrifying `@handle.platform_specific` we see everywhere.

In fact, I don't even see any reason to abstract the naming here: any POSIX system specifics will expect a file descriptor, and Windows system specifics will expect a HANDLE.

Trying to abstrtact this is actually confusing and misleading. We could have the false impression that `LibC.tcsetattr(@handle.platform_specific, ...)` will just work on Windows for example.

Reply:
I think you're right that it's bad naming, but I don't think that `#to_unsafe` is a good way to handle this. It's way too magic to pass a crystal object and it magically gets converted into a file descriptor. I think automatically calling `to_unsafe` is way too magic in general...

I think you're probably right about naming it `fd` on posix and `handle` on windows.